specforge/modeling/target/eagle3_target_model.py

```diff
@@ -182,38 +216,43 @@ def from_pretrained(
         device: str = None,
         cache_dir: Optional[str] = None,
         trust_remote_code: bool = False,
+        args: Optional[argparse.Namespace] = None,
         **kwargs,
     ) -> "SGLangEagle3TargetModel":
-        tp_size = dist.get_world_size(get_tp_group())
-        server_args = ServerArgs(
-            model_path=pretrained_model_name_or_path,
-            trust_remote_code=trust_remote_code,
-            dtype=torch_dtype,
-            enable_return_hidden_states=True,
-            disable_cuda_graph=True,
-            enable_torch_compile=True,
-            tp_size=tp_size,
-            pp_size=1,
-            attention_backend="torch_native",
-        )
+        if args is None:
+            server_args = ServerArgs(
+                model_path=pretrained_model_name_or_path,
+                trust_remote_code=trust_remote_code,
+                dtype=torch_dtype,
+                enable_return_hidden_states=True,
+                disable_cuda_graph=True,
+                enable_torch_compile=True,
+                tp_size=get_target_tp_size(),
+                pp_size=1,
+                attention_backend="torch_native",
+                mem_fraction_static=0.4,
+            )
+            target_micro_batch_size = None
+        else:
+            server_args = ServerArgs.from_cli_args(args)
```

Comment:
I recommend that we don't do this, because this might cause some confusion to the users.
1. Some arguments in SGLang are for optimizations other than prefill, these options won't take effect even if the user specify them
2. this will make `--help` of our training script extremely long.

I recommend that we only keep those important to prefill.